liblocation/src/main/java/com/mapbox/android/core/location/GoogleLocationEngine.java

```diff
@@ -170,4 +169,22 @@ private void connect() {
   private void updateRequestPriority(LocationRequest request) {
     REQUEST_PRIORITY.get(priority).update(request);
   }
-}
+
+  private void generatePendingIntent(Context context) {
+    pendingIntent = LocationPendingIntentProvider.buildIntent(context, this).retrievePendingIntent();
+  }
+
+  private void setupBroadcastReceiver(Context context) {
+    final String action = "GoogleLocationEngineBroadcast-" + this.toString();
```

Comment:
I think using just `hashCode` is safer, in case `toString` is overwritten at any time in the future. This value can also be stored as the field and passed to the intent provider.